src/bootstrap-table.js

```diff
@@ -3112,14 +3135,14 @@ class BootstrapTable {
     const row = this.data[index]
     const $tr = this.$body.find(Utils.sprintf('> tr[data-index="%s"][data-has-detail-view]', index))
 
-    if ($tr.next().is('tr.detail-view')) {
-      return
-    }
-
     if (this.options.detailViewIcon) {
       $tr.find('a.detail-icon').html(Utils.sprintf(this.constants.html.icon, this.options.iconsPrefix, this.options.icons.detailClose))
     }
 
+    if ($tr.next().is('tr.detail-view')) {
```

Comment:
Why are we moving this?

Reply:
The idea of moving this is to be able to call `expandRow()` in the cases where we already copied the whole details pane html. In this case expandRow will just switch the details icon to "minus" and return. Otherwise we would either have an expanded row with a "plus" icon, or we'd have to copy the icon switching code somewhere else.